### PR TITLE
Document OpenID Provider configuration

### PR DESCRIPTION
**Pull request by thoumasd**
The objective of this PR is to document how Opendatasoft supports the OpenID Connect protocol and how a domain administrator can configure the connection to an OpenID Connect provider.

To do so, a reorganization of the domain security configuration section was necessary.

I also used this opportunity to try to improve / simplify the SAML configuration documentation.

Below are the main changes added to this PR:
- Added a new OIDC configuration section.
- Move some documentation content out of the SAML configuration documentation: local / linked user accounts, default login page.
- Removed from SAML documentation references to "account mapper" and "attributes mapper" which IMHO were useless.
- Get rid of IdP acronym and use Identity Provider instead.
- Get rid of the "users federation terminology". Simplified to some extent by replaceing by "single sign-on", which should be what users are looking for.

[ch24112]

### CODE REVIEW

File: source/configuring_domain/01_managing_security/saml.rst [hunk 1]
```diff
@@ -1,94 +1,33 @@
...
 
 .. image:: images/configuration_SAML.png
-    :alt: SAML IDP configuration interface
+    :alt: SAML identity provider configuration interface
 
-3. Paste your identity provider metadata document in the "IDP metadata document" field.
+3. Paste your identity provider metadata document in the "Identity provider metadata document" field.
 
-4. If you are using Microsoft Azure Active Directory as an IDP, check the related checkbox.
+4. If you're identity provider is Microsoft Azure Active Directory, check the related checkbox.
```
Comment (olivier-peres) [inline source/configuring_domain/01_managing_security/saml.rst:22]: you're ⟶ your

**Review by NTerpo — APPROVED** [review]
👌🏻

File: source/configuring_domain/01_managing_security/oidc.rst [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+6. If you would like to use JWT tokens to impersonate API calls on the Opendatasoft platform, check the "Enable direct authentication with JWT bearer tokens" checkbox.
+
+7. List optional additional scopes
+
+   Scopes in OpenID Connect are used to define the set of information (claims) an application would like to receive about a connected user.
+   By default, Opendatasoft requires the following scopes: openid, profile and email. These scopes make it possible to authenticate a user
+   and to retrieve their email, first name and lastname.
+
+   You can require additional scopes to add claims as additional attributes in a user's profile and use them in
```
[the remark this answers is not in the document]
Comment (thoumasd) [reply source/configuring_domain/01_managing_security/oidc.rst:50]: You're right, I'll use "request" instead.

[the remark this answers is not in the document]
Comment (thoumasd) [reply source/configuring_domain/01_managing_security/oidc.rst:50]: Actually not, I'll use "use" instead. Good catch.

**Comment by thoumasd** [comment]
Thanks for your review @NTerpo @olivier-peres, updates pushed.

@remilejeune this PR is ready to be pushed.

**Comment by olivier-peres** [comment]
@thoumasd I think your edit ⟶ check change went wrong: commit 64341cf85c99796dcfd10a7b7f14cc1796db76fa applies it in places where `check` is correct and not where it was supposed to be applied.

**Pull request merged**